torchmetrics/functional/audio/sdr.py

```diff
@@ -0,0 +1,189 @@
+            loss to train neural separation networks.
+        filter_length:
+            The length of the distortion filter allowed
+        zero_mean:
+            When set to True, the mean of all signals is subtracted prior to computation of the metrics
+        load_diag:
+            If provided, this small value is added to the diagonal coefficients of
+            the system metrics when solving for the filter coefficients.
+            This can help stabilize the metric in the case where some of the reference
+            signals may sometimes be zero
+
+    Raises:
+        ValueError:
+            If ``fast-bss-eval`` package is not installed
+
+    Returns:
+        sdr value of shape ``[...]`` if compute_permutation is False, else ``[..., spk]``
+
+    Example:
+        >>> from torchmetrics.functional.audio import sdr
+        >>> import torch
+        >>> g = torch.manual_seed(1)
+        >>> preds = torch.randn(8000)
+        >>> target = torch.randn(8000)
+        >>> sdr(preds, target)
+        tensor(-12.0589)
+        >>> # use with pit
+        >>> from torchmetrics.functional.audio import pit
+        >>> preds = torch.randn(4, 2, 8000)  # [batch, spk, time]
+        >>> target = torch.randn(4, 2, 8000)
+        >>> best_metric, best_perm = pit(preds, target, sdr, 'max')
+        >>> best_metric
+        tensor([-11.6375, -11.4358, -11.7148, -11.6325])
+        >>> best_perm
+        tensor([[1, 0],
+                [0, 1],
+                [1, 0],
+                [0, 1]])
+
+    References:
+        [1] Vincent, E., Gribonval, R., & Fevotte, C. (2006). Performance measurement in blind audio source separation.
+         IEEE Transactions on Audio, Speech and Language Processing, 14(4), 1462–1469.
+        [2] Scheibler, R. (2021). SDR -- Medium Rare with Fast Computations.
+        [3] https://github.com/fakufaku/fast_bss_eval
+    """
+
+    if not _FAST_BSS_EVAL_AVAILABLE:
+        raise ValueError(
+            "SDR metric requires that fast-bss-eval is installed."
+            "Either install as `pip install torchmetrics[audio]` or `pip install fast-bss-eval`"
+        )
+    _check_same_shape(preds, target)
+
+    if not preds.dtype.is_floating_point:
+        preds = preds.float()  # for torch.norm
+
+    # half precision support
+    if preds.dtype == torch.float16:
+        preds = preds.to(torch.float32)
+
+    if preds.dtype != target.dtype:  # for torch.linalg.solve
+        target = target.to(preds.dtype)
+
+    if zero_mean:
+        preds = preds - preds.mean(dim=-1, keepdim=True)
+        target = target - target.mean(dim=-1, keepdim=True)
+
+    # normalize along time-axis
+    if not _TORCH_GREATER_EQUAL_1_8:
+        # use numpy if torch<1.8
+        warnings.warn(
+            "pytorch is under 1.8, thus SDR numpy version is used."
+            "For better performance and differentiability, you should change to pytorch 1.8+"
+        )
```

Comment:
please use `from torchmetrics.utilities import rank_zero_warn
```suggestion
        rank_zero_warn(
            "Pytorch is under 1.8, thus SDR numpy version is used."
            "For better performance and differentiability, you should change to Pytorch v1.8 or above."
        )
```